x-pack/plugins/endpoint/server/plugin.ts

```diff
@@ -3,28 +3,56 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
-
 import { Plugin, CoreSetup } from 'kibana/server';
 import { addRoutes } from './routes';
+import { PluginSetupContract as FeaturesPluginSetupContract } from '../../features/server';
 
 export type EndpointPluginStart = void;
 export type EndpointPluginSetup = void;
-export interface EndpointPluginSetupDependencies {} // eslint-disable-line @typescript-eslint/no-empty-interface
-
 export interface EndpointPluginStartDependencies {} // eslint-disable-line @typescript-eslint/no-empty-interface
 
+export interface EndpointPluginSetupDependencies {
+  features: FeaturesPluginSetupContract;
+}
+
 export class EndpointPlugin
   implements
     Plugin<
-      EndpointPluginStart,
       EndpointPluginSetup,
-      EndpointPluginStartDependencies,
-      EndpointPluginSetupDependencies
+      EndpointPluginStart,
+      EndpointPluginSetupDependencies,
+      EndpointPluginStartDependencies
     > {
-  public setup(core: CoreSetup) {
+  public setup(core: CoreSetup, plugins: EndpointPluginSetupDependencies) {
+    plugins.features.registerFeature({
+      id: 'endpoint',
+      name: 'Endpoint',
+      icon: 'bug',
+      navLinkId: 'endpoint',
+      app: ['endpoint', 'kibana'],
+      privileges: {
+        all: {
+          api: ['resolver'],
+          savedObject: {
+            all: [],
```

Comment:
**hint** I see there's a `ui` capability here that indicates you may be saving _something_ in the future. When that happens, be sure to update the saved object privileges here with the specific types you need to read and write. We're more than happy to help with this!